### PR TITLE
feat: HQM-54 staked and vested balances

### PR DESCRIPTION
**Pull request by iGroza**
Fixes #

## Proposed Changes

  -  HQM-54 staked and vested balances

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/components/locked-tokens.tsx [hunk 1]
```diff
@@ -10,15 +10,15 @@ import {Icon, IconButton, IconsName, Spacer, Text} from './ui';
 
 export interface LockedTokensProps {
   totalAmout: number;
-  availableAmout: number;
-  lockedAmout: number;
+  availableAmount: number;
```
Comment (vivalaakam) [inline src/components/locked-tokens.tsx:13]: Balance

File: src/models/staking-metadata.ts [hunk 2]
```diff
@@ -51,6 +51,7 @@ export class StakingMetadata extends Realm.Object {
           type: StakingMetadataType.delegation,
           delegator,
           validator,
+          // TODO: remove WEI
           amount: parseInt(amount, 10) / WEI,
```
Comment (vivalaakam) [inline src/models/staking-metadata.ts:55]: use here raw value aka hex without dividing

File: src/screens/locked-tokens.tsx [hunk 3]
```diff
@@ -4,32 +4,48 @@ import {LockedTokens} from '@app/components/locked-tokens';
...
+  const stakingBalances = useWalletsStakingBalance(visible);
 
-  const totalAmout = useMemo(
+  const availableAmount = useMemo(
     () =>
       Object.values(balances).reduce(
-        (prev, curr) => prev + curr.toNumber(),
+        (prev, curr) => prev + (curr?.toNumber() ?? 0),
         0,
       ) / WEI,
```
Comment (vivalaakam) [inline src/screens/locked-tokens.tsx:20]: use Balance without toNumber

File: src/components/account-info/account-info-header.tsx [hunk 4]
```diff
@@ -6,30 +6,45 @@ import {Color} from '@app/colors';
...
 
+import {StackedVestedTokens} from '../stacked-vested-tokens';
+
 const CARD_WIDTH = 78;
 const CARD_RADIUS = 8;
 
 export type AccountInfoProps = {
   wallet: Wallet;
-  balance: Balance;
+  balance: Balance | undefined;
```
Comment (ragozin-nikita) [inline src/components/account-info/account-info-header.tsx:30]: And down to `onSend`
```suggestion
  balance?: Balance;
```

File: src/components/stacked-vested-tokens.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import {Color} from '@app/colors';
+import {createTheme} from '@app/helpers';
+import {I18N} from '@app/i18n';
+import {Balance} from '@app/types';
+
+import {DashedLine} from './dashed-line';
+import {Icon, IconButton, IconsName, Spacer, Text} from './ui';
+
+export interface StackedVestedTokensProps {
+  unvestedBalance: Balance | undefined;
```
Comment (ragozin-nikita) [inline src/components/stacked-vested-tokens.tsx:14]:
```suggestion
  unvestedBalance?: Balance;
```

File: src/components/stacked-vested-tokens.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  lockedBalance: Balance | undefined;
+  vestedBalance: Balance | undefined;
+  stakingBalance: Balance | undefined;
+  balance: Balance | undefined;
+  onPressInfo(): void;
```
Comment (ragozin-nikita) [inline src/components/stacked-vested-tokens.tsx:19]:
```suggestion
  onPressInfo: () => void;
```

File: src/components/wallet-card.tsx [hunk 7]
```diff
@@ -15,18 +15,19 @@ import {
...
 import {Wallet} from '@app/models/wallet';
 import {Balance} from '@app/types';
 import {IS_IOS, SHADOW_COLOR_1, SYSTEM_BLUR_2} from '@app/variables/common';
 
 export type BalanceProps = {
   testID?: string;
   wallet: Wallet;
-  balance: Balance;
-  lockedTokensAmount: number;
+  balance: Balance | undefined;
```
Comment (ragozin-nikita) [inline src/components/wallet-card.tsx:28]:
```suggestion
  balance?: Balance;
```

**Review by ragozin-nikita — APPROVED** [review]

**Pull request merged**